sync_diff_inspector/report/report.go

```diff
@@ -0,0 +1,382 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package report
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/olekukonko/tablewriter"
+	"github.com/pingcap/errors"
+	"github.com/pingcap/log"
+	"github.com/pingcap/tidb-tools/pkg/dbutil"
+	"github.com/pingcap/tidb-tools/sync_diff_inspector/chunk"
+	"github.com/pingcap/tidb-tools/sync_diff_inspector/config"
+	"github.com/pingcap/tidb-tools/sync_diff_inspector/source/common"
+	"github.com/pingcap/tidb-tools/sync_diff_inspector/utils"
+	"go.uber.org/zap"
+)
+
+const (
+	// Pass means all data and struct of tables are equal
+	Pass = "pass"
+	// Fail means not all data or struct of tables are equal
+	Fail  = "fail"
+	Error = "error"
+)
+
+// ReportConfig stores the config information for the user
+type ReportConfig struct {
+	Host     string `toml:"host"`
+	Port     int    `toml:"port"`
+	User     string `toml:"user"`
+	Snapshot string `toml:"snapshot,omitempty"`
+	SqlMode  string `toml:"sql-mode,omitempty"`
+}
+
+// TableResult saves the check result for every table.
+type TableResult struct {
+	Schema      string                  `json:"schma"`
+	Table       string                  `json:"table"`
+	StructEqual bool                    `json:"struct-equal"`
+	DataSkip    bool                    `json:"data-skip"`
+	DataEqual   bool                    `json:"data-equal"`
+	MeetError   error                   `json:"-"`
+	ChunkMap    map[string]*ChunkResult `json:"chunk-result"` // `ChunkMap` stores the `ChunkResult` of each chunk of the table
+}
+
+// ChunkResult save the necessarily information to provide summary information
+type ChunkResult struct {
+	RowsAdd    int `json:"rows-add"`    // `RowAdd` is the number of rows needed to add
+	RowsDelete int `json:"rows-delete"` // `RowDelete` is the number of rows needed to delete
+}
+
+// Report saves the check results.
+type Report struct {
+	sync.RWMutex
+	Result       string                             // Result is pass or fail
+	PassNum      int32                              // The pass number of tables
+	FailedNum    int32                              // The failed number of tables
+	TableResults map[string]map[string]*TableResult `json:"table-results"` // TableResult saved the map of  `schema` => `table` => `tableResult`
+	StartTime    time.Time                          `json:"start-time"`
+	Duration     time.Duration                      `json:"time-duration"`
+	TotalSize    int64                              // Total size of the checked tables
+	SourceConfig [][]byte
+	TargetConfig []byte
```

Comment:
if you don't `json:"…"` those field will still be serialized with the original name. are these expected?